docs/src/modules/ROOT/pages/responding-to-change/responding-to-change.adoc

```diff
@@ -522,6 +522,109 @@ A `BestSolutionChangedEvent` does not guarantee that every `ProblemChange` has b
 
 . Use `Score.isSolutionInitialized()` instead of `Score.isFeasible()` to only ignore uninitialized solutions, but do accept infeasible solutions too.
 
+
+[#recommendedFitAPI]
+== Responding to adhoc changes
+
+With <<realTimePlanning,real-time planning>>, we can respond to a continuous stream of external changes.
+However, it is often necessary to respond to adhoc changes too,
+for example when a call center operator needs to arrange an appointment with a customer.
+In such cases, it is not necessary to use the full power of real-time planning.
+Instead, immediate response to the customer and a selection of available time windows are more important.
+This is where _Recommended Fit API_ comes in.
+
+The Recommended Fit API is a simple API that allows you to quickly respond to adhoc changes,
+while providing a selection of the best available options for fitting the change in the existing schedule.
+It doesn't use the full xref:optimization-algorithms/optimization-algorithms.adoc#localSearch[local search algorithm].
+Instead,
+it uses a simple xref:optimization-algorithms/optimization-algorithms.adoc#constructionHeuristics[greedy algorithm]
+together with xref:constraints-and-score/performance.adoc#incrementalScoreCalculation[incremental calculation].
+This combination allows the API to find the best possible fit within the existing solution in a matter of milliseconds,
+even for large planning problems.
+
+Once the customer has accepted of the available options
+and the change has been reflected in the solution,
+the full xref:optimization-algorithms/optimization-algorithms.adoc#localSearch[local search algorithm]
+can be used to optimize the entire solution around this change.
+This would be an example of <<continuousPlanning,continuous planning>>.
+
+[#usingRecommendedFitAPI]
+=== Using the Recommended Fit API
+
+The Recommended Fit API requires one uninitialized entity to be present in the solution:
+
+[source,java,options="nowrap"]
+----
+NurseRoster nurseRoster = ...; // Our planning solution.
+ShiftAssignment unassignedShift = new ShiftAssignment(...); // A new shift needs to be assigned.
+nurseRoster.getShiftAssignmentList().add(unassignedShift);
+----
+
+The `SolutionManager` is then used to retrieve the recommended fit for the uninitialized entity:
+
+[source,java,options="nowrap"]
+----
+SolutionManager<NurseRoster, HardSoftScore> solutionManager = ...;
+List<RecommendedFit<Employee, HardSoftScore>> recommendations =
```

Comment:
What happens if the planning entity has two planning variables of different types? Do we call the API twice (for every variable with different proposition functions)?

Reply:
Good point that I should mention that this requires a construction heuristic configuration in the solver config.